File: internal/core/info.go
package core

import (
	"fmt"
	"strings"
)

// InfoLine holds the counts displayed in the CLI info bar.
type InfoLine struct {
	Unprocessed int
	Looping     int
	Active      int
}

// GetInfoLine computes the info line data from the store.
// Unprocessed = count of todos with status "unprocessed"
// Looping = count of todos with stale_count >= 2
// Active = count of todos with status "active"
func GetInfoLine(store *Store) (InfoLine, error) {
	all, err := allTodos(store)
	if err != nil {
		return InfoLine{}, err
	}

	var info InfoLine
	for _, todo := range all {
		if todo.Status == "unprocessed" {
			info.Unprocessed++
		}
		if todo.Status == "active" {
			info.Active++
		}
		if todo.StaleCount >= 2 {
			info.Looping++
		}
	}
	return info, nil
}

// FormatInfoLine returns the formatted info line string for CLI display.
// If all counts are 0, returns an empty string.
func FormatInfoLine(info InfoLine) string {
	if info.Unprocessed == 0 && info.Looping == 0 && info.Active == 0 {
		return ""
	}
	var parts []string
	if info.Unprocessed > 0 {
		parts = append(parts, fmt.Sprintf("Unprocessed: %d", info.Unprocessed))
	}
	if info.Active > 0 {
		parts = append(parts, fmt.Sprintf("Active: %d", info.Active))
	}
	if info.Looping > 0 {
		parts = append(parts, fmt.Sprintf("Looping: %d", info.Looping))
	}
	return fmt.Sprintf("-- %s --", strings.Join(parts, " | "))
}

File: src/core/info.ts
// src/core/info.ts
//
// Info line computation and formatting.
// The info line is shown after every CLI command as a quick at-a-glance
// summary of the system state.

import type { Store } from './store.js';

export interface InfoCounts {
  unprocessed: number;
  active: number;
  looping: number;
}

/**
 * Fetches current info counts from the store.
 */
export function getInfoLine(store: Store): InfoCounts {
  return store.getInfoCounts();
}

/**
 * Formats info counts into a human-readable line.
 * Returns empty string if all counts are zero.
 * Otherwise: "-- Unprocessed: N | Active: M | Looping: K --"
 * Only sections with count > 0 are included, in fixed order.
 */
export function formatInfoLine(info: InfoCounts): string {
  if (info.unprocessed === 0 && info.active === 0 && info.looping === 0) {
    return '';
  }

  const parts: string[] = [];
  if (info.unprocessed > 0) parts.push(`Unprocessed: ${info.unprocessed}`);
  if (info.active > 0) parts.push(`Active: ${info.active}`);
  if (info.looping > 0) parts.push(`Looping: ${info.looping}`);

  return `-- ${parts.join(' | ')} --`;
}
